### PR TITLE
fix(scanner): Parse RH vuln names from RH links

### PR DESCRIPTION
**Pull request by jvdm**
## Description

Parsing RHEL vuln names in the updaters rather than using the V4 "mappers". This guarantees backward compatibility with Scanner V2. We don't want to fork away from ClairCore, but this fork is temporary as work to onboard new RH security data sources is in progress.

## Checklist
- [ ] Investigated and inspected CI test results
- [ ] Unit test and regression tests added
- [ ] Evaluated and added CHANGELOG entry if required
- [ ] Determined and documented upgrade steps
- [ ] Documented user facing changes (create PR based on [openshift/openshift-docs](https://github.com/openshift/openshift-docs) and merge into [rhacs-docs](https://github.com/openshift/openshift-docs/tree/rhacs-docs))

If any of these don't apply, please comment below.

### Here I tell how I validated my change

TODO(replace-me)  
Use this space to explain **how you validated** that **your change functions exactly how you expect it**.
Feel free to attach JSON snippets, curl commands, screenshots, etc. Apply a simple benchmark: would the information you
provided convince any reviewer or any external reader that you did enough to validate your change.

It is acceptable to assume trust and keep this section light, e.g. as a bullet-point list.

It is acceptable to skip testing in cases when CI is sufficient, or it's a markdown or code comment change only.  
It is also acceptable to skip testing for changes that are too taxing to test before merging. In such case you are
responsible for the change after it gets merged which includes reverting, fixing, etc. Make sure you validate the change
ASAP after it gets merged or explain in PR when the validation will be performed.  
Explain here why you skipped testing in case you did so.

Have you created automated tests for your change? Explain here which validation activities you did manually and why so.

### Reminder for reviewers

In addition to reviewing code here, reviewers **must** also review testing and request further testing in case the
performed one does not seem sufficient. As a reviewer, you must not approve the change until you understand the
performed testing and you are satisfied with it.

### CODE REVIEW

**Review by daynewlee — APPROVED** [review]
Looks ok to me but it might be better to have some manual test

File: scanner/updater/rhel/parser.go [hunk 1]
```diff
@@ -52,7 +52,22 @@ func (u *Updater) Parse(ctx context.Context, r io.ReadCloser) ([]*claircore.Vuln
...
+				if r.Source == "CVE" {
+					name = r.RefID
+					break
+				}
+			}
+		case ovalutil.RHBADefinition, ovalutil.RHSADefinition, ovalutil.RHEADefinition:
+			name = strings.TrimSpace(def.Title[:strings.Index(def.Title, ": ")])
+		default:
+			name = def.Title
+		}
```
Comment (RTann) [inline scanner/updater/rhel/parser.go:70]: to guarantee compatibility and to simplify this, you can mimic what scanner v2 does:

```
func name(definition oval.Definition) string {
	if len(definition.References) > 0 {
		return definition.References[0].RefID
	}

	return "" // <- maybe replace this with def.Title, though
}
```

File: scanner/mappers/mappers_test.go [hunk 2]
```diff
@@ -990,10 +990,18 @@ func Test_vulnerabilityName(t *testing.T) {
 			name:     "DSA-4591-1 cyrus-sasl2",
 			expected: "DSA-4591-1",
 		},
-		"RHEL": {
+		"RHEL/RHSA": {
 			name:     "RHSA-2023:0173: libxml2 security update (Moderate)",
 			expected: "RHSA-2023:0173",
 		},
+		"RHEL/RHBA": {
```
Comment (RTann) [inline scanner/mappers/mappers_test.go:997]: what would these have been prior to the update?

Comment (RTann) [reply scanner/mappers/mappers_test.go:997]: can you provide an example for what this fixes in the PR description?

**Review by RTann — APPROVED** [review]
Code LGTM, but I'm still not sure what the output was prior to this change. Can you provide an example?

**Pull request merged**